src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -414,7 +414,6 @@ protected internal virtual void OnProjectRemoved(ProjectId projectId)
             using (_serializationLock.DisposableWait())
             {
                 CheckProjectIsInCurrentSolution(projectId);
-                this.CheckProjectCanBeRemoved(projectId);
```

Comment:
Consider: if somebody else is overriding this to make a statement like "this workspace should never have projects removed ever" for sanity, this now broke them.